### PR TITLE
Add support to build Arm64 ISO for CBL-Mariner

### PR DESCRIPTION
**Pull request by chalamalasetty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

  Add support to build and generate Arm64 ISO for CBL-Mariner.
  Command to build Arm64 ISO:  sudo make iso CONFIG_FILE=./imageconfigs/full.json REBUILD_TOOLCHAIN=n REBUILD_PACKAGES=n REBUILD_TOOLS=y LOG_LEVEL=debug

###### Change Log  

- Change :  Add support to build Arm64 ISO for CBL-Mariner

###### Does this affect the toolchain?  

YES
**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/...

###### Test Methodology

- Pipeline build id: xxxx
   Built Arm64 ISO and X64 ISOs with these changes. Verified on HyperV setup, by booting with ISO, installed Mariner OS and successfully booted to Mariner OS.

### CODE REVIEW

File: toolkit/scripts/imggen.mk [hunk 1]
```diff
@@ -11,7 +11,11 @@ assets_files             = $(shell find $(assets_dir))
 imggen_local_repo        = $(MANIFESTS_DIR)/image/local.repo
 imagefetcher_local_repo  = $(MANIFESTS_DIR)/package/local.repo
 imagefetcher_cloned_repo = $(MANIFESTS_DIR)/package/fetcher.repo
+ifeq ($(shell uname -m),aarch64)
```
Comment (schmittjoseph) [inline toolkit/scripts/imggen.mk:14]: Can reference `$(build_arch)` variable here. Applies to below as well.

Comment (chalamalasetty) [reply toolkit/scripts/imggen.mk]: updated.

**Review by schmittjoseph — APPROVED** [review]

File: toolkit/resources/imageconfigs/packagelists/iso-initrd-packages-arm64.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
+{
+    "packages": [
```
Comment (PawelWMS) [inline toolkit/resources/imageconfigs/packagelists/iso-initrd-packages-arm64.json:2]: Are there any big differences between the AMD64 version? If there are non, let's use one JSON and if they have many packages in common, let's put the common part into one JSON and whatever differs into separate ones. This way it's going to be easier to understand the special requirements for each architecture.

File: toolkit/tools/isomaker/maker.go [hunk 3]
```diff
@@ -175,36 +179,55 @@ func (im *IsoMaker) setUpIsoGrub2Bootloader() {
...
 	logger.Log.Debug("Copying EFI modules into efiboot.img.")
-	// Copy Shim (bootx64.efi) and grub2 (grubx64.efi)
-	bootDirPath := filepath.Join(efiBootImgTempMountDir, "EFI", "BOOT")
-	bootx64EfiFilePath := filepath.Join(bootDirPath, "bootx64.efi")
-	im.extractFromInitrdAndCopy(bootx64BootloaderFile, bootx64EfiFilePath)
-	grubx64EfiFilePath := filepath.Join(bootDirPath, "grubx64.efi")
-	im.extractFromInitrdAndCopy(grubx64BootloaderFile, grubx64EfiFilePath)
-
+	// Copy Shim (boot<arch>64.efi) and grub2 (grub<arch>64.efi)
+	if runtime.GOARCH == "arm64" {
```
Comment (PawelWMS) [inline toolkit/tools/isomaker/maker.go:183]: There's a lot of code repetition here. Let's extract these two cases to a single function with arguments for the boot and Grub2 EFI files.

File: toolkit/tools/isomaker/maker.go [hunk 4]
```diff
@@ -175,36 +179,55 @@ func (im *IsoMaker) setUpIsoGrub2Bootloader() {
...
+		bootEfiUsbFilePath := filepath.Join(im.buildDirPath, "efi/boot/bootaa64.efi")
+		im.extractFromInitrdAndCopy(bootaa64BootloaderFile, bootEfiUsbFilePath)
+		grubEfiUsbFilePath := filepath.Join(im.buildDirPath, "efi/boot/grubaa64.efi")
+		im.extractFromInitrdAndCopy(grubaa64BootloaderFile, grubEfiUsbFilePath)
+	} else {
+		bootEfiUsbFilePath := filepath.Join(im.buildDirPath, "efi/boot/bootx64.efi")
+		im.extractFromInitrdAndCopy(bootx64BootloaderFile, bootEfiUsbFilePath)
+		grubEfiUsbFilePath := filepath.Join(im.buildDirPath, "efi/boot/grubx64.efi")
+		im.extractFromInitrdAndCopy(grubx64BootloaderFile, grubEfiUsbFilePath)
+	}
```
Comment (PawelWMS) [inline toolkit/tools/isomaker/maker.go:230]: Similar comment as above: let's remove the repetition. The directory paths look identical, so we'd probably only need to provide different file names for the two EFI modules.

File: toolkit/resources/assets/isomaker/iso_root_arch-dependent_files/arm64/isolinux/isolinux.cfg [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# D-I config version 2.0
```
Comment (PawelWMS) [inline toolkit/resources/assets/isomaker/iso_root_arch-dependent_files/arm64/isolinux/isolinux.cfg:1]: In the summary for this PR please fill the last part describing how you validated the change.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: toolkit/tools/isomaker/maker.go (hunk 4 above)
Comment (PawelWMS) [reply toolkit/tools/isomaker/maker.go:230]: I've provided my suggestions in #378.

File: toolkit/resources/imageconfigs/packagelists/iso-initrd-packages-arm64.json (hunk 2 above)
Comment (chalamalasetty) [reply toolkit/resources/imageconfigs/packagelists/iso-initrd-packages-arm64.json:2]: There are differences like calamares, grub2-pc etc which are not available for Arm64 arch. grub2-pc is required for x64 BIOS based systems.

File: toolkit/resources/assets/isomaker/iso_root_arch-dependent_files/arm64/isolinux/isolinux.cfg (hunk 5 above)
Comment (chalamalasetty) [reply toolkit/resources/assets/isomaker/iso_root_arch-dependent_files/arm64/isolinux/isolinux.cfg:1]: Updated steps on how I validated changes.

**Review by PawelWMS — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**